diff --git a/dist/unstable/httpapi/HttpApiBuilder.js b/dist/unstable/httpapi/HttpApiBuilder.js
index 61dc51d7..1e05ee83 100644
--- a/dist/unstable/httpapi/HttpApiBuilder.js
+++ b/dist/unstable/httpapi/HttpApiBuilder.js
@@ -14,6 +14,7 @@ import * as Result from "../../Result.js";
 import * as Schema from "../../Schema.js";
 import * as Issue from "../../SchemaIssue.js";
 import * as Transformation from "../../SchemaTransformation.js";
+import * as Scope from "../../Scope.js";
 import * as ServiceMap from "../../ServiceMap.js";
 import * as Stream from "../../Stream.js";
 import * as UndefinedOr from "../../UndefinedOr.js";
@@ -283,7 +284,7 @@ function handlerToHttpEffect(group, endpoint, services, handler, isRaw) {
     }
     const response = yield* handler(request);
     return Response.isHttpServerResponse(response) ? response : yield* encodeSuccess(response);
-  })).pipe(Effect.withErrorReporting, Effect.catch(error => Effect.orDie(encodeError(error))), Effect.provideServices(services));
+  })).pipe(Effect.withErrorReporting, Effect.catch(error => Effect.orDie(encodeError(error))), Effect.provideServices(ServiceMap.omit(Scope.Scope)(services)));
 }
 function handlerToRoute(group, handler, services) {
   const endpoint = handler.endpoint;
diff --git a/dist/unstable/httpapi/HttpApiSchema.d.ts b/dist/unstable/httpapi/HttpApiSchema.d.ts
index 07fe995a..8ca5b3ce 100644
--- a/dist/unstable/httpapi/HttpApiSchema.d.ts
+++ b/dist/unstable/httpapi/HttpApiSchema.d.ts
@@ -159,6 +159,47 @@ export interface asMultipartStream<S extends Schema.Top> extends Schema.brand<S[
  * @since 4.0.0
  */
 export declare function asMultipartStream(options?: Multipart_.withLimits.Options): <S extends Schema.Top>(self: S) => asMultipartStream<S>;
+/**
+ * @since 4.0.0
+ */
+export interface JsonPayloadEncoding {
+    readonly _tag: "Json";
+    readonly contentType: string;
+}
+/**
+ * @since 4.0.0
+ */
+export interface FormUrlEncodedPayloadEncoding {
+    readonly _tag: "FormUrlEncoded";
+    readonly contentType: string;
+}
+/**
+ * @since 4.0.0
+ */
+export interface TextPayloadEncoding {
+    readonly _tag: "Text";
+    readonly contentType: string;
+}
+/**
+ * @since 4.0.0
+ */
+export interface Uint8ArrayPayloadEncoding {
+    readonly _tag: "Uint8Array";
+    readonly contentType: string;
+}
+/**
+ * @since 4.0.0
+ */
+export interface MultipartPayloadEncoding {
+    readonly _tag: "Multipart";
+    readonly mode: "buffered" | "stream";
+    readonly contentType: string;
+    readonly limits?: Multipart_.withLimits.Options | undefined;
+}
+/**
+ * @since 4.0.0
+ */
+export type PayloadEncoding = JsonPayloadEncoding | FormUrlEncodedPayloadEncoding | TextPayloadEncoding | Uint8ArrayPayloadEncoding | MultipartPayloadEncoding;
 /**
  * Marks a schema as a JSON payload / response.
  *
